### PR TITLE
modulesync 0.7.0

### PR DESCRIPTION
**Pull request by bastelfreak**

### CODE REVIEW

File: manifests/plugin.pp [hunk 1]
```diff
@@ -1,9 +1,15 @@
-define minecraft::plugin($plugin_name = $title, $source) {
+define minecraft::plugin(
+  $plugin_name  = $title,
+  $source       = ''
```
Comment (daenney) [inline manifests/plugin.pp]: Can we turn this into `undef`?

File: manifests/plugin.pp [hunk 2]
```diff
@@ -1,9 +1,15 @@
...
+) {
 
   if $plugin_name =~ /^.*\.jar$/ {
     fail("minecraft plugin name ${plugin_name} must not end in '.jar'")
   }
-  
+
+  if $source == '' {
```
Comment (daenney) [inline manifests/plugin.pp]: `if !$source` ...

**Comment by daenney** [comment]
Can we sqooshy squishy squashy this a bit? It's 9 commits now with some duplicate commit messages and stuff.

**Comment by bastelfreak** [comment]
I'm on it, I just have to rage a bit about software, takes some time because there is so much software

File: manifests/plugin.pp [hunk 3]
```diff
@@ -1,9 +1,12 @@
-define minecraft::plugin($plugin_name = $title, $source) {
+define minecraft::plugin(
+  $source,
+  $plugin_name  = $title
+) {
 
   if $plugin_name =~ /^.*\.jar$/ {
     fail("minecraft plugin name ${plugin_name} must not end in '.jar'")
```
Comment (igalic) [inline manifests/plugin.pp]: this seems silly
we could probably just normalize it…

File: manifests/plugin.pp [hunk 4]
```diff
@@ -1,9 +1,12 @@
...
   }
-  
+
   wget::fetch { $plugin_name:
```
Comment (igalic) [inline manifests/plugin.pp]: this should probably be replaced with archive

File: manifests/plugin.pp (hunk 3 above)
Comment (daenney) [inline manifests/plugin.pp]: Ya, we're already checking if it ends in `.jar`, if it doesn't just tack it on... But that should probably be part of a different PR.

File: manifests/plugin.pp (hunk 4 above)
Comment (daenney) [inline manifests/plugin.pp]: Yes, more PR's!

Comment (bastelfreak) [inline manifests/plugin.pp]: created an issue

File: manifests/plugin.pp (hunk 3 above)
Comment (bastelfreak) [inline manifests/plugin.pp]: created an issue
